File: app/src/main/java/com/sanketvagadiya/ui/Work/WorkViewModel.java
package com.sanketvagadiya.ui.Work;

import android.util.Log;

import androidx.lifecycle.ViewModel;

import com.sanketvagadiya.Constants.Constants;
import com.sanketvagadiya.Constants.GetOnlineData;
import com.sanketvagadiya.Model.WorkModel;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.ExecutionException;

public class WorkViewModel extends ViewModel {
    private List<WorkModel> workList = new ArrayList<>();

    public WorkViewModel() {
        getJson();
    }

    private JSONObject getJson() {
        JSONObject jsonObject = null;
        GetOnlineData getOnlineData = (GetOnlineData) new GetOnlineData(Constants.PAST_WORK_URL).execute();
        try {
            jsonObject = getOnlineData.get();
            if (jsonObject != null) {
                setJsonData(jsonObject);
            }
        } catch (ExecutionException e) {
            Log.e("err", "err" + e);
        } catch (InterruptedException e) {
            e.printStackTrace();
            Log.e("err", "err" + e);
        }
        return jsonObject;
    }

    private void setJsonData(JSONObject jsonObject) {
        try {
            JSONArray jsonArray = jsonObject.getJSONArray(Constants.PAST_WORK_JSONARRAY);

            for (int i = 0; i < jsonArray.length(); i++) {
                JSONObject jsonObj = jsonArray.getJSONObject(i);
                String roleName = jsonObj.getString(Constants.PAST_WORK_JSON_ROLE_NAME);
                String companyName = jsonObj.getString(Constants.PAST_WORK_JSON_COMPANY_NAME);
                String companyLocation = jsonObj.getString(Constants.PAST_WORK_JSON_LOCATION);
                String joinFrom = jsonObj.getString(Constants.PAST_WORK_JSON_JOIN_FROM);
                String joinTo = jsonObj.getString(Constants.PAST_WORK_JSON_JOIN_TO);
                String jobResponsibilities = jsonObj.getString(Constants.PAST_WORK_JSON_RESPONSIBILITY);

                String imageUrl = jsonObj.getString(Constants.PAST_WORK_JSON_IMAGEURL);

                workList.add(new WorkModel(roleName
                        , companyName, companyLocation, joinFrom
                        , joinTo, jobResponsibilities
                        , imageUrl
                ));
            }
            if (workList.size() > 0) {
                setWorkList(workList);
            }
        } catch (Exception e) {
            Log.e("WorkViewModel err", "err:" + e);
        }
    }

    public List<WorkModel> getWorkList() {
        return workList;
    }

    public void setWorkList(List<WorkModel> workList) {
        this.workList = workList;
    }

}

File: app/src/main/java/com/sanketvagadiya/ui/Work/WorkViewModel.kt
package com.sanketvagadiya.ui.Work

import android.util.Log

import androidx.lifecycle.ViewModel

import com.sanketvagadiya.Constants.Constants
import com.sanketvagadiya.Constants.GetOnlineData
import com.sanketvagadiya.Model.ProjectModel
import com.sanketvagadiya.Model.WorkModel

import org.json.JSONArray
import org.json.JSONObject

import java.util.ArrayList
import java.util.concurrent.ExecutionException

class WorkViewModel : ViewModel() {
    var workList = ArrayList<WorkModel>()

    private val json: JSONObject?
        get() {
            var jsonObject: JSONObject? = null
            val getOnlineData = GetOnlineData(Constants.PAST_WORK_URL).execute() as GetOnlineData
            try {
                jsonObject = getOnlineData.get()
                if (jsonObject != null) {
                    setJsonData(jsonObject)
                }
            } catch (e: ExecutionException) {
                Log.e("err", "err$e")
            } catch (e: InterruptedException) {
                e.printStackTrace()
                Log.e("err", "err$e")
            }

            return jsonObject
        }

    init {
        json
    }

    private fun setJsonData(jsonObject: JSONObject) {
        try {
            val jsonArray = jsonObject.getJSONArray(Constants.PAST_WORK_JSONARRAY)

            for (i in 0 until jsonArray.length()) {
                val jsonObj = jsonArray.getJSONObject(i)
                val roleName = jsonObj.getString(Constants.PAST_WORK_JSON_ROLE_NAME)
                val companyName = jsonObj.getString(Constants.PAST_WORK_JSON_COMPANY_NAME)
                val companyLocation = jsonObj.getString(Constants.PAST_WORK_JSON_LOCATION)
                val joinFrom = jsonObj.getString(Constants.PAST_WORK_JSON_JOIN_FROM)
                val joinTo = jsonObj.getString(Constants.PAST_WORK_JSON_JOIN_TO)
                val jobResponsibilities = jsonObj.getString(Constants.PAST_WORK_JSON_RESPONSIBILITY)

                val imageUrl = jsonObj.getString(Constants.PAST_WORK_JSON_IMAGEURL)

                workList.add(WorkModel(roleName, companyName, companyLocation, joinFrom, joinTo, jobResponsibilities, imageUrl
                ))
            }
            if (workList.size > 0) {
                setWorkList(workList)
            }
        } catch (e: Exception) {
            Log.e("WorkViewModel err", "err:$e")
        }

    }

    fun getWorkList(): List<WorkModel> {
        return workList
    }

    fun setWorkList(workList: List<WorkModel>) {
        this.workList = workList as ArrayList<WorkModel>



    }

}
